rust_src/src/symbols.rs

```diff
@@ -404,5 +404,42 @@ pub fn symbol_value(symbol: LispSymbolRef) -> LispObject {
     }
     val
 }
+/// Non-nil if VARIABLE has a local binding in buffer BUFFER.
+/// BUFFER defaults to the current buffer.
+#[lisp_fn(min = "1")]
+pub fn local_variable_p(variable: LispObject, buffer: LispBufferOrCurrent) -> bool {
+    let buf: LispBufferRef = buffer.into();
+    let mut symbol: LispSymbolRef = variable.into();
+
+    while symbol.is_alias() {
+        symbol = symbol.get_indirect_variable();
+    }
+
+    match symbol.get_redirect() {
+        symbol_redirect::SYMBOL_PLAINVAL => true,
+        symbol_redirect::SYMBOL_LOCALIZED => {
+            let blv = unsafe { symbol.get_blv() };
+            let variable: LispObject = symbol.into();
+            if blv.get_where().eq(buf) {
+                blv.found()
+            } else {
+                buf.local_vars_iter()
+                    .any(|local_var| variable.eq(local_var))
```

Comment:
`local_var` is itself a Cons. This should be:

    .any(|local_var| {
        let (car, _) = local_var.into();
        variable.eq(car)
    }